var/spack/repos/builtin/packages/lucene/package.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Lucene(Package):
+    """
+    Apache Lucene is a high-performance, full featured text search engine
+    library written in Java.
+    """
+
+    homepage = "https://lucene.apache.org/"
+    url      = "http://archive.apache.org/dist/lucene/java/8.3.1/lucene-8.3.1.tgz"
+    list_url = "http://archive.apache.org/dist/lucene/java/"
+    list_depth = 2
+
+    version('8.3.1', sha256='acd61ad458d16f3c98b9dd4653c6a34dd666a965842e461f7cdf8947fa041e1a')
+    version('8.3.0', sha256='67c4f8081f24ff9f4eb4f2b999ac19f7a639b416e5b6f1c1c74e0524a481fc7e')
+    version('8.2.0', sha256='505cad34698b217fd6ceee581a8215223a47df5af820c94ca70a6bdbba9d5d7c')
+    version('8.1.1', sha256='d62b0acdf2b1ed7a25ccdb593ad8584caeaa20cc9870e22790d3ec7fa6240a8c')
+
+    def url_for_version(self, version):
+        url = "http://archive.apache.org/dist/lucene/java/{0}/lucene-{0}.tgz"
+        return url.format(version)
```

Comment:
You don’t actually need this, Spack should be about to handle this url automatically.

Reply:
Fixed it in https://github.com/spack/spack/pull/14316/commits/17558d11c2943e55b9ae390401cfa097798e360d